### ISSUE TITLE
ukb-to-bids converter complains about source_id from zipfile

### ISSUE THREAD

**Issue by dlq** [aramis-lab/clinica#783]
```
[dlq@beluga2 20227]$ uname -r
3.10.0-1160.76.1.el7.x86_64
```

```
[dlq@beluga2 20227]$ clinica --version
clinica, version 0.7.2
```

```
[dlq@beluga2 20227]$ clinica convert ukb-to-bids 10 ../../Tabular/670041 .
2022-11-08 14:57:33,225:INFO:All clinical data have been found
/home/dlq/.local/lib/python3.8/site-packages/clinica/iotools/converters/ukb_to_bids/ukb_utils.py:86: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only.
  dataframe = pd.concat([dataframe_nifti, dataframe_dicom], 0)
Traceback (most recent call last):
  File "/home/dlq/.local/bin/clinica", line 8, in <module>
    sys.exit(main())
  File "/home/dlq/.local/lib/python3.8/site-packages/clinica/cmdline.py", line 89, in main
    cli()
  File "/home/dlq/.local/lib/python3.8/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/dlq/.local/lib/python3.8/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/dlq/.local/lib/python3.8/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/dlq/.local/lib/python3.8/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/dlq/.local/lib/python3.8/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/dlq/.local/lib/python3.8/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/dlq/.local/lib/python3.8/site-packages/clinica/iotools/converters/ukb_to_bids/ukb_to_bids_cli.py", line 29, in cli
    convert_images(dataset_directory, bids_directory, clinical_data_directory)
  File "/home/dlq/.local/lib/python3.8/site-packages/clinica/iotools/converters/ukb_to_bids/ukb_to_bids.py", line 30, in convert_images
    imaging_data = read_imaging_data(path_to_dataset)
  File "/home/dlq/.local/lib/python3.8/site-packages/clinica/iotools/converters/ukb_to_bids/ukb_utils.py", line 98, in read_imaging_data
    split_zipfile = split_zipfile.astype({"source_id": "int64"})
  File "/home/dlq/.local/lib/python3.8/site-packages/pandas/core/generic.py", line 5887, in astype
    raise KeyError(
KeyError: "Only a column name can be used for the key in a dtype mappings argument. 'source_id' not found in columns."
```

**Comment by MatthieuJoulot** [comment]
Hi @dlq,

It seems that `ukb-to-bids` cannot find any data in your dataset, or filtered it all. I would agree that if it is the case, this error isn't user friendly at all, and something we would definitely like to change.

In order to make sure I'm understanding correctly, could you give me a little context of what you are trying to do ?
What would be most helpful would be if you could give me the modality and format (dicom or nifti) of the data you are trying to convert !

**Comment by dlq** [comment]
Sorry for the delay… I’m trying to convert a directory of bulk 20227 zip files to BIDS.  So NIfTI fMRI files as input.

***@***.*** 20227]$ ls 10
1000011_20227_2_0.zip  1036042_20227_2_0.zip  1066257_20227_2_0.zip
1000011_20227_3_0.zip  1036042_20227_3_0.zip  1066511_20227_2_0.zip
1000083_20227_2_0.zip  1036087_20227_2_0.zip  1066544_20227_2_0.zip
1000139_20227_2_0.zip  1036211_20227_2_0.zip  1066544_20227_3_0.zip
1000261_20227_2_0.zip  1036275_20227_2_0.zip  1066651_20227_2_0.zip
1000347_20227_2_0.zip  1036306_20227_2_0.zip  1066747_20227_2_0.zip
1000394_20227_2_0.zip  1036404_20227_2_0.zip  1066802_20227_2_0.zip
1000514_20227_2_0.zip  1036540_20227_2_0.zip  1067159_20227_2_0.zip
1000537_20227_2_0.zip  1036661_20227_2_0.zip  1067297_20227_2_0.zip
1000635_20227_2_0.zip  1036676_20227_2_0.zip  1067388_20227_2_0.zip
1000635_20227_3_0.zip  1036724_20227_2_0.zip  1067418_20227_2_0.zip
1000791_20227_2_0.zip  1036790_20227_2_0.zip  1067430_20227_2_0.zip
1000996_20227_2_0.zip  1036846_20227_2_0.zip  1067451_20227_2_0.zip
1001074_20227_2_0.zip  1036878_20227_2_0.zip  1067451_20227_3_0.zip
1001115_20227_2_0.zip  1036939_20227_2_0.zip  1067531_20227_2_0.zip
1001159_20227_2_0.zip  1036943_20227_2_0.zip  1067921_20227_2_0.zip
...

**Comment by MatthieuJoulot** [comment]
Okay, thanks for the reply ! We have chosen to convert the fMRI data from UK Biobank using the dicoms and not the niftis ([explanation here](https://aramislab.paris.inria.fr/clinica/docs/public/latest/Converters/UKBtoBIDS/)) which is why the code errors: the data doesn't correspond to our criteria...

**Comment by NicolasGensollen** [comment]
Hi @dlq 
The error message was improved in #785 and should make it clear that there is something wrong with the input data. Furthermore, the comment from @MatthieuJoulot above explains why you end up in this situation.
This issue was closed automatically when merging #785, but feel free to re-open it if you think that it still needs discussions.
